api/apps/api/src/modules/access-control/projects-acl/project-acl.service.ts

```diff
@@ -92,4 +94,109 @@ export class ProjectAclService implements ProjectAccessControl {
       this.canViewSolutionRoles,
     );
   }
+
+  async checkUserIsOwner(userId: string, projectId: string): Promise<Permit> {
+    return !!(await this.roles.findOne({
+      where: {
+        projectId,
+        userId,
+        roleName: Roles.project_owner,
+      },
+    }));
+  }
+
+  async checkLastOwner(userId: string, projectId: string): Promise<Permit> {
+    const allOwnersInProject = await this.roles.find({
+      where: {
+        projectId,
+        roleName: Roles.project_owner,
+        userId: Not(userId),
+      },
+    });
+    return allOwnersInProject.length >= 1;
+  }
+  /**
+   * @debt This module should not involve user details and it should deal with it
+   * using a standalone module that will access the data just to read it. We
+   * have to get back to it once scenarios, organizations and solutions are included
+   * inside the access-module.
+   */
+  async findUsersInProject(
+    projectId: string,
+    userId: string,
+  ): Promise<UsersProjectsApiEntity[] | Permit> {
```

Comment:
Consider returning `Either<SomeErrorSymbols | ProjectUsers[]`

Also: we shouldn't leak entities to controller to avoid unwanted interactions (why would controller need to know about entity and make actions on it?)